go/exorelay/main.go

```diff
@@ -57,3 +66,42 @@ func (exoRelay *ExoRelay) Send(eventName string, payload map[string]interface{})
 
 	return websocket.Message.Send(exoRelay.socket, serializedBytes)
 }
+
+func (exoRelay *ExoRelay) receiveMessage(message structs.Message) {
+	if exoRelay.messageHandlerMapping == nil {
+		return
+	}
+	if exoRelay.messageHandlerMapping[message.Name] == nil {
+		return
+	}
+	exoRelay.messageHandlerMapping[message.Name](message.Payload)
+}
+
+func (exoRelay *ExoRelay) listenForMessages() {
+	for {
+		message, err := exoRelay.readMessage()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			fmt.Println("Error reading message from websocket:", err)
+			break
+		} else {
+			exoRelay.receiveMessage(message)
+		}
+	}
+}
+
+func (exoRelay *ExoRelay) readMessage() (structs.Message, error) {
+	var bytes []byte
+	if err := websocket.Message.Receive(exoRelay.socket, &bytes); err != nil {
+		return structs.Message{}, err
+	}
+
+	var unmarshaled structs.Message
+	err := json.Unmarshal(bytes, &unmarshaled)
+	if err != nil {
+		return structs.Message{}, err
+	}
+
+	return unmarshaled, nil
+}
```

Comment:
`readMessage` is an exact duplicate of what lives in exocom-mock. Because `websocket` is a dependency, I get an error about different types as the definition for the socket being passed is in separate places.

Thus to remove this duplication, I think we would need to encapsulate that entire library (something I attempted to do this morning and then threw out as I don't think its worth it. Could still refactor down the road.